source/postprocess/particle_density_histogram.cc

```diff
@@ -0,0 +1,268 @@
+              Create a table with the actual distribution
+              Take distance between "perfect" table and the actual table
+              return the ratio between the observed distance and the worst case distance.
+              a value of 1 represents the worst possible distribution, a value of 0 represents a completely even distribution.
+            */
+            unsigned int particles_in_cell = 0;
+            for (unsigned int particle_manager_index = 0; particle_manager_index < this->n_particle_managers(); ++particle_manager_index)
+            {
+              particles_in_cell += this->get_particle_manager(particle_manager_index).get_particle_handler().n_particles_in_cell(cell);
+            }
+            if(particles_in_cell > 0)
+            {
+              cells_with_particles++;
+              double particles_in_cell_double = static_cast<double>(particles_in_cell);
+              double granularity_double = static_cast<double>(granularity);
+              double ideal_n_particles_per_bucket = particles_in_cell_double/(granularity_double*granularity_double);
+
+              /*
+              buckets_ideal contains doubles because the ideal
+              number of particles per bucket will not always be an integer.
+              */
+              Table<dim,double> buckets_ideal; 
+              TableIndices<dim> bucket_sizes;
+              for (unsigned int i=0; i<dim; ++i){
+                bucket_sizes[i] = granularity;
+              }
+              buckets_ideal.reinit(bucket_sizes);
+              double bucket_width = 1.0/granularity_double;
+              buckets_ideal.fill(ideal_n_particles_per_bucket);
+
+              Table<dim,unsigned int> buckets_actual; 
+              buckets_actual.reinit(bucket_sizes);
+              sortParticles(cell,buckets_actual,bucket_width);
+              /*
+              in the worst case, all particles are in one bucket. 
+              (granularity*dim)-1 is equal to the number of buckets
+              in the table minus 1 (the bucket all the particles are in)
+              */
+              double worst_case_error_squared = 
+              ((particles_in_cell_double-ideal_n_particles_per_bucket)*(particles_in_cell_double-ideal_n_particles_per_bucket))+
+              ((ideal_n_particles_per_bucket*ideal_n_particles_per_bucket)*((granularity_double*static_cast<double>(dim))-1.0));
+              
+              double actual_error_squared = 0;
+              for (unsigned int x=0; x<granularity; x++){
+                for (unsigned int y=0; y<granularity; y++){
+                  TableIndices<dim> entry_index;
+                  entry_index[0] = x;
+                  entry_index[1] = y;
+                  //do another loop if in 3d
+                  if (dim == 3){
+                    for (unsigned int z=0; z<granularity; z++){                   
+                        entry_index[2] = z;
+                    }
+                  }                  
```

Comment:
this doesnt work for 3D at the moment. The loop above would complete, before you actually fill any bucket in the z-direction.